src/filter_tree/filter_tree.c

```diff
@@ -169,25 +180,41 @@ int _applyPredicateFilters(const FT_FilterNode *root, const Record r) {
 }
 
 int FilterTree_applyFilters(const FT_FilterNode *root, const Record r) {
-	/* Handle predicate node. */
-	if(IsNodePredicate(root)) {
-		return _applyPredicateFilters(root, r);
-	}
+	switch(root->t) {
+	case FT_N_COND: {
+		/* root->t == FT_N_COND, visit left subtree. */
+		int pass = FilterTree_applyFilters(LeftChild(root), r);
 
-	/* root->t == FT_N_COND, visit left subtree. */
-	int pass = FilterTree_applyFilters(LeftChild(root), r);
+		if(root->cond.op == OP_AND && pass == 1) {
+			/* Visit right subtree. */
+			pass *= FilterTree_applyFilters(RightChild(root), r);
+		}
 
-	if(root->cond.op == OP_AND && pass == 1) {
-		/* Visit right subtree. */
-		pass *= FilterTree_applyFilters(RightChild(root), r);
-	}
+		if(root->cond.op == OP_OR && pass == 0) {
```

Comment:
I think this would be better as an `else if`

Reply:
👍